estate/models/estate_property.py

```diff
@@ -159,3 +164,15 @@ def action_mark_canceled(self):
             record._set_offer(0.0, "")
             record.state = "canceled"
         return True
+
+    @api.constrains("selling_price", "expected_price")
+    def _check_selling_price(self):
+        for record in self:
+            # although it is recommended to use float_is_zero, in this case is
+            # important to check that the selling_price is strictly bigger than
+            # zero, not nearly zero
+            if record.selling_price > 0.0:
+                if float_compare(record.selling_price, 0.9 * record.expected_price, precision_digits=2) == -1:
```

Comment:
Use a new variable for `0.9 * record.expected_price`

Why 0.9?

Reply:
In the exercise of [this](https://www.odoo.com/documentation/16.0/developer/tutorials/getting_started/11_constraints.html#python) section, they ask

 >  Add a constraint so that the selling price cannot be lower than 90% of the expected price.
 
 From there comes the `0.9`.